.circleci/config.yml

```diff
@@ -36,18 +35,26 @@ jobs:
 
       # Then, build the actual app in a container shared between Linux and Windows
       - run: docker pull livepeerci/build:latest || echo 'no pre-existing cache found'
-      - run: docker build --build-arg HIGHEST_CHAIN_TAG -t livepeerci/build:latest --cache-from=livepeerci/build:latest -f docker/Dockerfile.build .
+      - run: |-
+          source ./ci_env.sh
+          docker build --build-arg HIGHEST_CHAIN_TAG -t livepeerci/build:latest --cache-from=livepeerci/build:latest -f docker/Dockerfile.build .
       - run: docker push livepeerci/build:latest
 
-      # Finally, build the minimal go-livepeer distributable
+      # Finally, build the minimal go-livepeer distributable. We publish two tags for each build:
+      # livepeer/go-livepeer:BRANCH_NAME and livepeer/go-livepeer:VERSION_STRING. Both are useful
+      # to pull from in different contexts.
       - run: |-
           # Our Docker tag name should be our branch name with just alphanums
-          TAG=$(echo $CIRCLE_BRANCH | tr -cd '[:alnum:]_')
-          docker build -t livepeer/go-livepeer:${TAG}-linux -f docker/Dockerfile.release-linux .
-          docker push livepeer/go-livepeer:${TAG}-linux
-          # Manifest step is optional in case the Windows build hasn't finished yet
-          docker manifest create livepeer/go-livepeer:${TAG} livepeer/go-livepeer:${TAG}-linux livepeer/go-livepeer:${TAG}-windows || true
-          docker manifest push livepeer/go-livepeer:${TAG} || true
+          BRANCH_TAG=$(echo $CIRCLE_BRANCH | sed 's/\//-/g' | tr -cd '[:alnum:]_-')
+          VERSION_TAG=$(./print_version.sh)
+          docker build -t current-build -f docker/Dockerfile.release-linux .
```

Comment:
why build the image with the temporary tag, "current-build" ?   How do we ensure that "current-build" tag doesn't persist but the branch tag & version tags do?

Reply:
I added functionality here to push to both e.g. `livepeer/go-livepeer:mainnet` and `livepeer/go-livepeer:0.5.3` during the build. `current-build` is just a placeholder variable; on line 52 I apply the actual tag then push it. `current-build` doesn't persist anywhere - it's not pushed, so it's just part of the ephemeral local Docker context.

Reply:
thanks for explaining :+1: